### DISCUSSION TITLE
Does ArgoCD supports ADO Integration with ServicePrincipals instead of PAT

### DISCUSSION THREAD

**Discussion by omerfsen** [argoproj/argo-cd#14185] [General]
I see on https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Generators-SCM-Provider/#azure-devops it uses a PAT .. I

Is there a way to use SP on ArgoCD ...https://devblogs.microsoft.com/devops/introducing-service-principal-and-managed-identity-support-on-azure-devops/ 

 https://learn.microsoft.com/en-us/azure/devops/integrate/get-started/authentication/service-principal-managed-identity?view=azure-devops

As you know PAT have an expiry date etc.. So it will be like using Github Application instead of Bot User's  PAT on Github

**Comment by jannfis** [comment]
Hi. No, Argo CD doesn't support AD Service Principals or other vendor-native authentication mechanisms for accessing Git repositories.

It's is incredible difficult to maintain and support authentication mechanisms for vendors that are not freely available (such as GitHub or GitLab) and require serious effort or money to get a reliable ("forever") test environment to be used in continuous integration etc.

**Comment by thikade** [comment]
Hi. need the same - so has this changed since 06/2023?
